binding.gyp

```diff
@@ -1,26 +1,28 @@
 {
-  "targets": [
-    {
-      "target_name": "w32appcontainertokens",
-      "cflags!": [
-        "-fno-exceptions"
-      ],
-      "cflags_cc!": [
-        "-fno-exceptions"
-      ],
-      "sources": [
-        "src/native.cpp"
-      ],
-      "include_dirs": [
-        "<!@(node -p \"require('node-addon-api').include\")"
-      ],
-      "libraries": [],
-      "dependencies": [
-        "<!(node -p \"require('node-addon-api').gyp\")"
-      ],
-      "defines": [
-        "NAPI_DISABLE_CPP_EXCEPTIONS"
-      ]
-    }
-  ]
+    "targets": [
+        {
+            "target_name": "w32appcontainertokens",
+            "msvs_configuration_attributes": {"SpectreMitigation": "Spectre"},
+            "msvs_settings": {
+                "VCCLCompilerTool": {
+                    "ExceptionHandling": 1,
+                    "AdditionalOptions": [
+                        "/guard:cf",
+                        "/sdl",
+                        "/w34244",
+                        "/we4267",
+                        "/ZH:SHA_256",
+                    ],
+                },
+                "VCLinkerTool": {"AdditionalOptions": ["/guard:cf"]},
+            },
+            "cflags!": ["-fno-exceptions"],
+            "cflags_cc!": ["-fno-exceptions"],
+            "sources": ["src/native.cpp"],
+            "include_dirs": ["<!@(node -p \"require('node-addon-api').include\")"],
+            "libraries": [],
+            "dependencies": ["<!(node -p \"require('node-addon-api').gyp\")"],
+            "defines": ["NAPI_DISABLE_CPP_EXCEPTIONS"],
```

Comment:
compiler flags are indicating to build with exceptions enabled but we are mixing a different mode from `node-addon-api`. I see that the change is a port of existing code, but nevertheless we should address since we are in this file.

You can refer to https://github.com/microsoft/vscode-node-sqlite3/commit/61db6fcfac21dcf9708a7e0e74e3a988f0134d11 for example.